### ISSUE TITLE
Localization entries are not applied completely

### ISSUE THREAD

**Issue by ChristianPervoelz** [mapeditor/tiled#3443]
**Describe the bug**
Not all translations are applied immediately after switching the language.

**To Reproduce**
For the following steps it is assumed the ui language is set to 'English'.
1. Start Tiled
2. Open Preferences menu (Edit > Preferences)
3. Switch to 'Interface'
4. Select a language different than 'English' (best one with an (almost) complete translation)

Result: Most texts are changed, but not all. (See image below)

**Expected behavior**
All texts are changed.

**Media**
![tiled_directlyAfterSettingLanguage](https://user-images.githubusercontent.com/24943047/184158224-c40a1979-2eb7-4416-a521-621c725fdb82.png)
Left side: How it looks directly after changing the language
Right side: After restarting Tiled with the correct display

**Specifications:**
 - OS: Windows 11
 - Tiled Version: 1.9.0

**Comment by ChristianPervoelz** [comment]
This is of course a total minor-minor-minor bug, as usually users don't change languages too often ;)
(Exception might be, when a language different than English is used and the user wants to create an issue here on GitHub, it might happen one changes the language to make it easier for maintainers)

**Comment by bjorn** [comment]
Yes, some of these texts are a little hard to adjust dynamically, but thanks for highlighting these, I'll have a look at fixing this eventually. :-)

**Comment by SIDDHAARTHAA** [comment]
Hey @bjorn, I’d like to work on this 🙂

I tried handling `LanguageChange` for `PropertiesWidget` and `NoTilesetWidget`, re-translating `Custom Properties`, and checking German `unfinished` entries (`Map Size`, `Output Chunk Size`, etc.). Some labels still don’t update immediately without restart.

Could you point me to the intended pattern for runtime language updates here?

**Comment by bjorn** [comment]
@SIDDHAARTHAA That `LanguageChange` event is the right thing to handle, you can find a lot of examples in the source code. If you're having issues updating strings in specific cases, feel free to ask additional questions.

This bug is a little outdated because the Properties view no longer has these "Property" and "Value" columns, but as you've noticed there are other strings affected by this problem that you can have a look at.

The "Template Editor" title is already set in a `TemplatesDock::retranslateUi` function, but the problem is that this function is currently only called from the constructor and not in response to the `LanguageChange` event, like other such functions.

**Comment by SIDDHAARTHAA** [comment]
I added missing runtime retranslation hooks (`QEvent::LanguageChange` / `Preferences::languageChanged`) in several places, which fixed some cases but not all.

There seem to be two separate issues:
1. Some locale entries are `unfinished`, so English fallback is expected.
2. Some dynamic UI/model/property texts are created once and not consistently refreshed on language change, so they stay stale until UI/model recreation (often restart).

So string-by-string fixes don’t scale; this likely needs a centralized retranslation approach for dynamic texts.

**Comment by bjorn** [comment]
> 1. Some locale entries are `unfinished`, so English fallback is expected.

Yes, this is because I've only last week updated the translation files for the upcoming 1.12 release, which received about 100 new strings to translate. So even translations which were complete in Tiled 1.11.2 will need a significant update. This is why Tiled 1.12 is in "string freeze" now, so that we may release with some of the translations being complete.

> So string-by-string fixes don’t scale; this likely needs a centralized retranslation approach for dynamic texts.

Hmm, dynamic re-translation based on `QEvent::LanguageChange` already works in most cases, and it would help if you would mention in which cases it does not. To me it really seems like something that needs to be looked into on a case-by-cases basis.

I'm not sure what a "centralized retranslation approach" would look like, but even then I think we'd still need to make sure each translatable string takes part in this system. There's no way to magically automate this.

**Comment by SIDDHAARTHAA** [comment]
Got it, that makes sense regarding the unfinished translations.

The cases I was referring to are ones where the string exists but the UI does not update after the LanguageChange event. For example the Template Editor title (TemplatesDock::retranslateUi is only called in the constructor).

I'll go through the remaining cases where widgets/models don't refresh on LanguageChange and handle them individually.
